hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/MessageTimeout.java

```diff
@@ -0,0 +1,47 @@
+package pl.allegro.tech.hermes.frontend.publishing;
+
+import org.xnio.XnioExecutor;
+import pl.allegro.tech.hermes.frontend.publishing.message.Message;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+class MessageTimeout {
+
+    private final String messageId;
+    private volatile XnioExecutor.Key timeoutKey;
+    private volatile Message message;
+    private volatile Optional<Consumer<Void>> timeoutConsumer = Optional.empty();
+
+    public MessageTimeout(String messageId) {
+        this.messageId = messageId;
+    }
+
+    public String getMessageId() {
+        return messageId;
+    }
+
+    public Message getMessage() {
+        return message;
+    }
+
+    public void setMessage(Message message) {
+        this.message = message;
+    }
+
+    public boolean remove() {
+        return timeoutKey.remove();
+    }
+
+    public void setTimeoutKey(XnioExecutor.Key timeoutKey) {
```

Comment:
This can be passed in a constructor. This setter is always called after constructor and timeoutKey is not optional.